gulpfile.js

```diff
@@ -20,45 +21,88 @@ const styles = () => {
       autoprefixer()
     ]))
     .pipe(sourcemap.write("."))
-    .pipe(dest("source/css"))
+    .pipe(dest("dist/css"))
     .pipe(sync.stream());
 }
 
 exports.styles = styles;
 
-// Logo
+// HTML
 
-const logo = () => {
-  return src("source/logo/*.svg")
+const html = () => {
+  return src("source/*.html")
+    .pipe(htmlmin({
+      collapseWhitespace: true
+    }))
+    .pipe(dest("dist"));
+}
+
+// Images
+
+const images = () => {
+  return src("source/img/**/*.{png,jpg,svg}")
     .pipe(imagemin([
+      imagemin.mozjpeg({
+        progressive: true
+      }),
+      imagemin.optipng({
+        optimizationLevel: 3
+      }),
       imagemin.svgo()
     ]))
-    .pipe(dest("source/icons"))
+    .pipe(dest("source/img"))
+}
+
+exports.images = images;
+
+// Logo
+
+const logo = () => {
+  return src("source/img/logo/*.svg")
+    .pipe(dest("dist/icons"))
 }
 
 exports.logo = logo;
 
 // Svg stack
 
 const svgstack = () => {
-  return src("source/img/**/*.svg")
+  return src("source/icons/*.svg")
     .pipe(svgsprite({
       mode: {
         stack: {}
       }
     }))
     .pipe(rename("stack.svg"))
-    .pipe(dest("source/icons"));
+    .pipe(dest("dist/img"));
 }
 
 exports.svgstack = svgstack;
 
+// Copy
+
+const copy = (done) => {
+  src([
+    "source/fonts/*.{woff2,woff}",
+    "source/img/*.ico",
```

Comment:
favicon.ico должен лежать в корне сервера, то есть в dist. Для этого и в исходниках надо не в подпапки его положить.
```suggestion
    "source/*.ico",
```